components/expression/expression-renderer.js

```diff
@@ -179,16 +179,31 @@ ExpressionRenderer.prototype.drawMatrixCells_ = function() {
     width: this.heatmapWidth_,
     height: this.heatmapHeight_
   }]);
-  cells.enter().append('rect')
-    .classed('cell', true);
+  var heatmapData = this.dataReady_();
+  console.log(heatmapData);
+  var colorScale = d3.scale.linear()
+      .domain([heatmapData.valueMin, heatmapData.valueMax])
+      .range(['white', 'red']);
+  var heatmapRow = cells.data(heatmapData.values)
+      .enter().append('g')
+      .attr("transform", "translate(" + 30 + "," + 10 + ")");
+  var heatmapRects = heatmapRow.selectAll('.rect')
+      .data(function(d) { return d; })
+      .enter().append('rect')
+      .attr('width', 5)
```

Comment:
width and height shall be determined by the current heatmap size, in order to fill up the entire heatmap area.